homeassistant/components/airtouch4/config_flow.py

```diff
@@ -0,0 +1,56 @@
+"""Config flow for AirTouch4."""
+from airtouch4pyapi import AirTouch
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_HOST
+
+from .const import DOMAIN
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+
+
+async def _validate_connection(hass: core.HomeAssistant, host):
+    airtouch = AirTouch(host)
+    await airtouch.UpdateInfo()
+
+    if hasattr(airtouch, "error"):
+        if isinstance(airtouch.error, Exception):
+            raise airtouch.error
+        raise ConnectionError()
+    return bool(airtouch.GetGroups())
+
+
+class AirtouchConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle an Airtouch config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+        errors = {}
+
+        host = user_input[CONF_HOST]
+
+        try:
+            result = await _validate_connection(self.hass, host)
+            if not result:
+                errors["base"] = "no_units"
+        except (OSError, ConnectionError):
```

Comment:
It's better if the library handles exceptions and raises library specific exceptions. Ie the library handles OSError and raises a custom exception if needed.